web/unit_tests/test.sh

```diff
@@ -0,0 +1,92 @@
+#! /bin/bash
+
+# A simple utility script to facilitate our different modes for unit-testing KMW.
+# It's rigged to be callable by NPM to facilitate testing during development when in other folders.
+
+display_usage ( ) {
+    echo "test.sh [-CI | -debug | -? | -h | -help] [-reporter <reporter>]"
+    echo
+    echo "  -CI               to run unit tests in CI mode on BrowserStack."
+    echo "                    This script requires your credentials to be set in environment variables - see "
+    echo "                    https://stackoverflow.com/questions/32450546/hiding-browserstack-key-in-karma"
+    echo ""
+    echo "  -debug            to establish a Karma server that facilitates unit test debugging"
+    echo "                    Not compatible with -CI."
+    echo ""
+    echo "  -reporter         sets the test engine to utilize the specified <reporter>."
+    echo "                    Valid options:  BrowserStack, teamcity, dots, progress, mocha"
+    echo ""
+    echo "  -? | -h | -help   to display this help information"
+    echo ""
+    echo "                    Specifying no option will perform a simple, single run of the test cases on"
+    echo "                    the dominant set of browsers for the currently-detected active OS."
+    echo ""
+    exit 0
+}
+
+# Designed to determine which set of browsers should be available for local testing,
+# based upon the current system OS.
+get_config_OS_id ( ) {
+  # Subject to change with future improvements.
+  if [[ "${OSTYPE}" = "darwin"* ]]; then
+    config_os_id="mac"
+  elif [[ "${OSTYPE}" = "msys" ]]; then
+    config_os_id="win"
+  elif [[ "${OSTYPE}" = "cygwin" ]]; then
+    config_os_id="win"
+  fi
+  
+  # We mostly care about platform-independent browser coverage with Linux;
+  # this will look for just Chrome and Firefox.
+  if [ -z $OSTYPE ]; then
+    config_os_id="linux"
+  fi
+}
```

Comment:
You should combine these `ifs` and guarantee `config_os_id` gets set to something.

On my Ubuntu 12.04 machine, `${OSTYPE} == linux-gnu`, and karma fails trying to use a conf file `"unit_tests/manual..conf.js"`